lib/scope.js

```diff
@@ -122,6 +124,13 @@ AST_Toplevel.DEFMETHOD("figure_out_scope", function(options){
                 scope.uses_eval = save_scope.uses_eval;
                 scope.directives = save_scope.directives;
             }
+            if (options.safari10) {
+                if (node instanceof AST_For ||
+                    node instanceof AST_ForIn ||
+                    node instanceof AST_ForOf) {
```

Comment:
Since:
```
var AST_ForOf = DEFNODE("ForOf", null, {
    $documentation: "A `for ... of` statement",
}, AST_ForIn);
```
you can remove the `AST_ForOf` condition.